### ISSUE TITLE
New-CmAzIaasVm doesn't allow you to specify the disk caching policy

### ISSUE THREAD

**Issue by eliot-trustm** [Trustmarque/cloudmarque-azure-pwsh#121]
managed disks on virtual machines have various settings for disk caching which are not exposed in the cmdlet.

**Comment by nathankitchen** [comment]
Add support for disk caching in VMs the New-CmAzIaasVm cmdlet. In the virtualmachines configuration section there should be additional settings for caching, which will require moving the dataDisks node to be an array of a complex type rather than just sizes (note, some settings cut for brevity, see full spec [here](https://docs.trustmarque.com/cloudmarque/reference/commands/new-cmaziaasvm.html)).

``` yaml
    virtualMachines:     # Mandatory
      - name: "string"    # Mandatory
        size: "string"    # Mandatory
        osDisk:     # Mandatory
          osType: "Windows"    # Mandatory
          expandableDiskSizeGB: 0    # Mandatory
        dataDiskSizes:     # Optional
          - 0
```

This will need to be modified to the following:

``` yaml
    virtualMachines:     # Mandatory
      - name: "string"    # Mandatory
        size: "string"    # Mandatory
        osDisk:     # Mandatory
          osType: "Windows"    # Mandatory
          expandableDiskSizeGB: 0    # Mandatory
          caching: "ReadOnly"
          writeAccelerator: true
        dataDisks:     # Optional
          - sizeGB: 20
            caching: "ReadOnly"
            writeAccelerator: true
```

The `caching` setting should be:
- optional
- enum ("None", "ReadOnly", "ReadWrite")
- default to `None`
- example should be set to `None`

The `writeAccelerator` setting should be:
- optional
- boolean
- default to `false`
- example should be set to `false`

**Comment by deepend-dev** [comment]
# Write Accelerator is parked for now because of below restrictions - 
https://docs.microsoft.com/en-us/azure/virtual-machines/how-to-enable-write-accelerator

## Restrictions when using Write Accelerator

- Costly M-Series Virtual Machines (VMs)
- Premium storage
- Requires Azure Managed Disks which we currently do not support with VM implementation.

**Comment by deepend-dev** [comment]
## Disk Caching is implemented successfully.

**Comment by tm-luke** [comment]
Fixed  in 1.0.10
